src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/NameQuotedValuePairListEncoding.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.Designer;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug
+{
+
+    [Export(typeof(INameValuePairListEncoding))]
+    [ExportMetadata("Encoding", "NameQuotedValuePairListEncoding")]
+    internal sealed class NameQuotedValuePairListEncoding : INameValuePairListEncoding
+    {
+        public IEnumerable<(string Name, string Value)> Parse(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                yield break;
+            }
+
+            foreach (var entry in ReadEntries(value))
+            {
+                var (entryKey, entryValue) = SplitEntry(entry);
+
+                if (!string.IsNullOrEmpty(entryKey))
+                {
+                    yield return (DecodeCharacters(entryKey), StripQuotes(DecodeCharacters(entryValue)));
+                }
+            }
+
+            static string DecodeCharacters(string value)
+            {
+                return value.Replace("/=", "=").Replace("/,", ",").Replace("//", "/").Replace("/\"", "\"");
+            }
+        }
+
+        public string Format(IEnumerable<(string Name, string Value)> pairs)
+        {
+            return string.Join(",", pairs.Select(pair => $"{EncodeCharacters(pair.Name)}=\"{EncodeCharacters(pair.Value)}\""));
+
+            static string EncodeCharacters(string value)
+            {
+                int i = value.Length - 1;
+                while( i > -1)
+                {
+                    if (value[i] == '/' || value[i] == ',' || value[i] == '=' || value[i] == '"')
+                    {
+                        if (i == 0 || value[i-1] != '/')
+                        {
+                            value = value.Insert(i, "/");
+                        }
+                        else if(value[i - 1] == '/')
+                        {
+                            i--;
+                        }
+                    }
+                    i--;
+                }
+                return value;
+            }
+        }
+
+        private static string StripQuotes(string value)
```

Comment:
nit: `StringQuotes`, `ReadEntries` and `SplitEntry` methods are only called in one place, so could be converted to a local function of that caller, alongside the other local function the caller already has.